storage/badger/operation/common.go

```diff
@@ -53,6 +53,36 @@ func insert(key []byte, entity interface{}) func(*badger.Txn) error {
 	}
 }
 
+// upsert will encode the given entity using JSON and
+// if the key doesn't exist it will insert the resulting binary data in the badger DB under the provided key.
+// otherwise it will update the value under the provided key.
+func upsert(key []byte, entity interface{}) func(*badger.Txn) error {
```

Comment:
Does it need to actually be an upsert? It seems like we will insert the same value each time, we just may insert it twice under some circumstances. In this case we can use `operation.SkipDuplicates`, which is basically: "insert, but don't throw an error if it already exists"

Reply:
Aha, I didn't know SkipDuplicates exists 👍 , I'll use it then instead